pkg/kv/kvserver/logstore/sideload.go

```diff
@@ -84,8 +84,14 @@ func MaybeSideloadEntries(
 
 	cow := false
 	for i := range entriesToAppend {
-		if !SniffSideloadedRaftCommand(entriesToAppend[i].Data) {
-			otherEntriesSize += int64(len(entriesToAppend[i].Data))
+		ent := &entriesToAppend[i]
+		if ent.Type != raftpb.EntryNormal {
+			otherEntriesSize += int64(len(ent.Data))
+			continue
+		}
+		v, cmdID, data := kvserverbase.DecodeRaftCommand(ent.Data) // cheap
+		if v != kvserverbase.RaftVersionSideloaded {
+			otherEntriesSize += int64(len(ent.Data))
```

Comment:
How about a helper here?

```
func isSideloaded(entry) {
	v := ...
	return entry.Type == EntryNormal && v == RaftVersionSideloaded
}
```

Basically, why not just fix the `Sniff` helper (by passing the entry to it, rather than its data field)?